src/os/bluestore/BlockDevice.h

```diff
@@ -89,7 +89,8 @@ class BlockDevice {
 
   virtual int read(uint64_t off, uint64_t len, bufferlist *pbl,
 	   IOContext *ioc, bool buffered) = 0;
-  virtual int read_buffered(uint64_t off, uint64_t len, char *buf) = 0;
+  virtual int read(uint64_t off, uint64_t len, char *buf,
+           bool buffered) = 0;
```

Comment:
note that in the current implementation these are slightly different... the read() variant assumes reads are block-aligned, whereas read_buffered can do byte-granuarltiy length and offset.  Not sure if this is a problem or not...